pyfr/plugins/fwh.py

```diff
@@ -75,20 +81,18 @@ def _init_surf(self, ele_map):
             # Get face operators
             itype, proj, norm = eles.basis.faces[fidx]
 
-            # Create a quadratue on suface
+            # Create a quadrature on suface
             qpts, qwts = self._surf_quad(itype, proj)
             norm = np.array([norm for x in qpts])
 
-            # Get phyical location, transformations, and normals
+            # Get physical location, transformations, and normals
             ploc = eles.ploc_at_np(qpts)[..., eidxs]
-            rcpdjac = eles.rcpdjac_at_np(qpts)[..., eidxs]
             pnorm = eles.pnorm_at(qpts, norm)[:, eidxs]
 
             # Get ops and components of the surface
             m0 = eles.basis.ubasis.nodal_basis_at(qpts)
-            norm_mag = np.linalg.norm(pnorm, axis=-1, keepdims=True)
-            n = (pnorm / norm_mag).transpose(2, 0, 1).reshape(self.ndims, -1)
-            area = (qwts[:, None] / rcpdjac).reshape((-1))
+            n = pnorm.transpose(2, 0, 1).reshape(self.ndims, -1)
+            area = (qwts[:, None]*np.ones_like(eidxs)).reshape(-1)
```

Comment:
This is still wasteful.  We are just repeating an array dozens of time, wasting memory and memory bandwidth.  Just store the weights and broadcast when they are actually used.